source/agora/node/FullNode.d

```diff
@@ -137,7 +136,7 @@ public class FullNode : API
         this.metadata = this.getMetadata(config.node.data_dir);
 
         this.config = config;
-        this.params = params;
+        this.params = new immutable(ConsensusParams)(config.node.validator_cycle);
```

Comment:
Wait this doesn't make any sense. Why is this code in `FullNode` instead of `Validator`?

Reply:
Ah never mind.. I was confused about the name.

Reply:
At first we had `validator_cycle` in the node config. It was supposed to be configurable by each validator node.

Then we decided we will have a hardcoded validator cycle for all validators.

Then we wanted to make it flexible for unittests, so we moved this constant into `ConsensusParams`.

And now we moved back again to step 1, where the validator cycle is configurable via the config.

So my questions is:

- Shouldn't we have just deleted `validator_cycle` from the config? It's supposed to only live in the `ConsensusParams` if it's supposed to be a constant

Reply:
The thing that I missed originally was that we might also want to apply those changes to a system integration test which is not compiled with `unittest`, hence why we have some consensus-critical values in the config (we have Genesis block too).

Reply:
Oh right, system integration. Hmm..

So then consensus parameters will always live in the node config?

Reply:
IOW would it make sense for these consensus params to just become a ConsensusConfig struct that lives inside of NodeConfig?